Core/Core/Offline/OfflineService.swift

```diff
@@ -47,3 +47,22 @@ public final class OfflineServiceLive: OfflineService {
         !availabilityService.status.isConnected
     }
 }
+
+class OfflineServiceModel: ObservableObject {
+    @Published var isOffline: Bool = false
+
+    private let networkAvailabilityService = NetworkAvailabilityServiceLive()
+    private var subscriptions = Set<AnyCancellable>()
+
+    init(offlineService: OfflineServiceLive) {
+        unowned let unownedSelf = self
+        networkAvailabilityService.startMonitoring()
+        networkAvailabilityService
+            .startObservingStatus()
+            .receive(on: DispatchQueue.main)
+            .sink {
+                unownedSelf.isOffline = $0 == .disconnected && offlineService.isOfflineModeEnabled()
+            }
+            .store(in: &subscriptions)
```

Comment:
`offlineService.isOfflineModeEnabled` already checks the network condition and the feature flag settings so this logic is not needed here.

Reply:
true, done.